pkg/alertmanager/local/store.go

```diff
@@ -0,0 +1,71 @@
+package local
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alerts"
+	"github.com/prometheus/alertmanager/config"
+)
+
+// StoreConfig configures a static file alertmanager store
+type StoreConfig struct {
+	Path string `yaml:"path"`
+}
+
+// RegisterFlags registers flags related to the alertmanager file store
+func (cfg *StoreConfig) RegisterFlags(f *flag.FlagSet) {
+	f.StringVar(&cfg.Path, "alertmanager.storage.local.path", "", "Path at which alertmanager configurations are stored.")
+}
+
+// Store is used to load user alertmanager configs from a local disk
+type Store struct {
+	cfg StoreConfig
+}
+
+// NewStore returns a new file alert store.
+func NewStore(cfg StoreConfig) (*Store, error) {
+	return &Store{cfg}, nil
+}
+
+// ListAlertConfigs returns a list of each users alertmanager config.
+func (f *Store) ListAlertConfigs(ctx context.Context) (map[string]alerts.AlertConfigDesc, error) {
+	configs := map[string]alerts.AlertConfigDesc{}
+	err := filepath.Walk(f.cfg.Path, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		ext := filepath.Ext(info.Name())
+
+		if !info.IsDir() && (ext == ".yml" || ext == ".yaml") {
```

Comment:
As previously mentioned, I would reverse this into a guard. Looks easier to read the code, if we `return nil` in case it's not a file or not a YAML extension.

I would also suggest to log with a `Warn` any skipped entry, because we wouldn't expect any spurious file/directory within `f.cfg.Path`, right?

Reply:
Sorry I didn't rectify this previously